### PR TITLE
[GPU] Slice operation

### PR DESCRIPTION
**Pull request by lznamens**
### Details:
- This PR is a copy of PR #8267 which was rebased on top of the master branch due to many changes in ov_gpu_lib code structure
 - Slice operation in cldnn_engine was implemented according to https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/movement/Slice_8.md
 - Slice functionality is covered by single layer tests

Ticket: *54648*

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableInputDataType(Datatype::INT32);
+    k.EnableInputDataType(Datatype::INT64);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableOutputDataType(Datatype::INT32);
+    k.EnableOutputDataType(Datatype::INT64);
+    k.EnableAllInputLayout();
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp:64]: It seems not true, seems that only bfyx and bfzyx are supported.

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    FillCLKernelData(kernel_data.kernels[0], dispatch_data, params.engineInfo,
+            kernelName, jit, entry_point);
+
+    return {kernel_data};
+}
+
+KernelsPriority SliceKernelRef::GetKernelsPriority(const Params&/*params*/,
+        const optional_params&/*options*/) const {
+    return FORCE_PRIORITY_1;
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp:51]: reference kernels used to be set as  DONT_USE_IF_HAVE_SOMETHING_ELSE

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+ParamsKey SliceKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp:56]: what about int8 ? the primitive has i8 and u8 in the impelementation_map

File: inference-engine/thirdparty/clDNN/tests/test_cases/slice.cpp [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+        auto gen = [&distrib, &mersenne_engine]() {
+            return static_cast<float>(distrib(mersenne_engine)) / std::numeric_limits<int>::max();
+        };
+        std::vector<float> result(size);
+        std::generate(std::begin(result), std::end(result), gen);
+        return result;
+    }
+
+    void TearDown() override {
+        layout data_layout ( data_types::f32, format::bfyx, tensor{input_shape_} );
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/tests/test_cases/slice.cpp:33]: please add TCs for other layouts too

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp (hunk 1 above)
Comment (ytorzuk-altran) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp:64]: Resolved in #8267

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp (hunk 2 above)
Comment (ytorzuk-altran) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp:51]: Resolved in #8267

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp (hunk 3 above)
Comment (ytorzuk-altran) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/slice/slice_kernel_ref.cpp:56]: Resolved in #8267

File: inference-engine/thirdparty/clDNN/tests/test_cases/slice.cpp (hunk 4 above)
Comment (ytorzuk-altran) [reply inference-engine/thirdparty/clDNN/tests/test_cases/slice.cpp:33]: Resolved in #8267

**Comment by lznamens** [comment]
Changes have been merged in 8267

**Pull request closed without merging**